### PR TITLE
Add v2 based on amphp/amp v3

### PR DESCRIPTION
**Pull request by iggyvolz**
This pull request should go into the v2 branch but I don't believe it's possible to create a pull request to a branch that doesn't exist.

### CODE REVIEW

File: src/PsrHttpClient.php [hunk 1]
```diff
@@ -35,4 +36,9 @@ public function request(PsrRequest $psrRequest, ?CancellationToken $cancellation
 
         return $this->psrAdapter->toPsrResponse($response);
     }
+
+    public function sendRequest(PsrRequest $request): PsrResponse
+    {
+        return $this->request($request);
+    }
```
Comment (kelunik) [inline src/PsrHttpClient.php]: I guess we can just rename the above method instead of adding a second one?

Comment (iggyvolz) [reply src/PsrHttpClient.php]: For some reason I thought the extra optional parameter would conflict - I'll fix that.

Comment (kelunik) [reply src/PsrHttpClient.php]: It doesn't as long as it has a default value. 🙂

**Comment by kelunik** [comment]
Thanks! I've created the branch and merged your changes. We can iterate on that. :-)

**Comment by enumag** [comment]
With v3 us there a reason not to implement it directly?

**Comment by kelunik** [comment]
Yes, we want a proper stream abstraction that works across all libraries, so an adapter on top instead of an adapter for our stream abstraction seems the better way to go. Especially with features such as HTTP/2 push, etc.

**Pull request merged**